tests/L0/run_transformer/test_pipeline_parallel_fwd_bwd.py

```diff
@@ -46,23 +70,21 @@ def _forward_backward_test_impl(
         fwd_bwd_func: FwdStepFunc,
         pipeline_model_parallel_world_size: Optional[int],
         virtual_pipeline_model_parallel_size: Optional[int],
+        dtype: Optional[torch.dtype] = torch.float32,
     ) -> None:
-        for dtype, deallocate_pipeline_outputs in itertools.product(
-            [torch.float32] + _get_autocast_dtypes(), (True, False),
-        ):
+        for deallocate_pipeline_outputs in (True, False):
```

Comment:
and IMHO leaving the second option is preferable because it shows me that the fwd-bwd function works with torch.float16 & torch.bfloat16. My major concerns with these two dtypes are (a) p2p communication using them and (b) `grad_scaler` for torch.half.

```suggestion
        for dtype, deallocate_pipeline_outputs in itertools.product(
            [torch.float32] + _get_autocast_dtypes(), (True, False),
        ):
```